src/collection.ts

```diff
@@ -725,4 +735,29 @@ export class Collection<T extends TigrisCollectionType> implements ICollection {
 
 		return clonedDocs;
 	}
+
+	private setCreatedAtForDocsIfNotExists(docs: Array<T>, createdAt: Date): Array<T> {
+		const collectionTarget = this._metadataStorage.collections.get(this.collectionName)?.target;
+		const collectionFields = this._metadataStorage.getCollectionFieldsByTarget(collectionTarget);
+
+		const collectionCreatedAtFieldNames = collectionFields.filter((field) => {
+			return field?.schemaFieldOptions?.timestamp === "createdAt";
+		}).map(f => f.name);
+
```

Comment:
Thanks for addressing the issue @akhill10 . We can cache the `collectionCreatedAtFieldNames` at instantiation in constructor. That will significantly improve the performance. We will be good to merge this then.

Reply:
Good suggestion. We can tackle that in a separate PR.